readthedocs/projects/tasks/builds.py

```diff
@@ -595,6 +596,16 @@ def execute(self):
         # ``__exit__``
         self.data.build_director.create_build_environment()
         with self.data.build_director.build_environment:
+
+            # NOTE: check if the build uses `build.commands` and only run those
+            if self.data.config.build.commands:
+                self.update_build(state=BUILD_STATE_BUILDING)
```

Comment:
I think we're 👍  on moving forward with this in some kind of beta feature. Either a feature flag, or some very large warning in the docs that this beta and isn't supported yet.

Reply:
I'm fine moving forward with this to gather some feedback.

Regarding how to communicate,

- I think using a feature flag may generate some extra work in our support queue that I'd like to avoid if possible
- Also, with a feature flag, fewer people will test this because the entry barrier is higher
- I like the idea of adding a warning in the docs communicating this is beta and is not supported yet, but I think we need something else on the build page as well
- I'd add a warning message on each build details page that uses `build.commands` too, to communicate this effectively to users that copy and paste the config from other projects without reading our docs 😄 

I'm 👍🏼 to add this without a feature flag + in beta status + warning in docs + warning in build page.